app/models/authentication_event.rb

```diff
@@ -0,0 +1,17 @@
+class AuthenticationEvent < ApplicationRecord
+  belongs_to :user, foreign_key: 'user_id'
+
+  validates :email_submitted, presence: true
+  validates :event_type, presence: true
+  validates :user_id, presence: true
+  
+  # Logs an email validation event
+  def log_email_validation(email, event_type, user_id = nil)
```

Comment:
The `log_email_validation` method does not handle potential exceptions that might occur during the creation of the `AuthenticationEvent`. Consider adding error handling to ensure the application can gracefully handle such scenarios.